harness/determined/exec/fetch_context.py

```diff
@@ -0,0 +1,30 @@
+import base64
+import distutils.util
+import io
+import os
+import tarfile
+
+from determined import constants
+from determined.common.api import certs, request
+
+if __name__ == "__main__":
+    exp_id = os.environ["DET_EXPERIMENT_ID"]
+    master_addr = os.environ["DET_MASTER_ADDR"]
+    master_port = os.environ["DET_MASTER_PORT"]
+    use_tls = distutils.util.strtobool(os.environ.get("DET_USE_TLS", "false"))
+
+    master_url = f"http{'s' if use_tls else ''}://{master_addr}:{master_port}"
+    certs.cli_cert = certs.default_load(master_url=master_url)
```

Comment:
if this preamble is going to appear in many startup scripts, it may be nice to library-ize it. a mistake here that only affects specific customer envs seems easy to write and it would be high impact.

Reply:
I agree.  I was going to take that on as part of the Generic API work, if that's ok with you.